src/os/im/mapping/ellipsemappingmanager.js

```diff
@@ -0,0 +1,268 @@
+
+
+/**
+ * Class for Ellipse Mapping Manager
+ * @unrestricted
+ */
+class MappingManager {
+  /**
+   * Constructor.
+   * @ngInject
+   * @param {*=} opt_source
+   */
+  constructor(opt_source) {
+    /**
+     * The listener added to the source
+     * @type {*}
+     * @private
+     */
+    this.sourceListener_;
+
+    /**
+     * The source used
+     * @type {*}
+     * @private
+     */
+    this.source_ = opt_source;
+
+    /**
+     * FeatureCount
+     * @type {number}
+     * @private
+     */
+    this.featureCount_ = opt_source ? this.source_.getFeatureCount() : undefined;
+
+    /**
+     * The mappings that were last applied to the data
+     * @type {Array<RenameMapping>}
+     * @private
+     */
+    this.appliedMappings_;
+
+    /**
+     * Mapping Options used for the ui
+     * @type {MappingOptions|undefined}
+     * @private
+     */
+    this.mappingOptions_ = undefined;
+  }
+
+  /**
+   * Create Mappings for Ellipse Data
+   * @param {MappingOptions} data
+   * @return {Array<RenameMapping>}
+   */
+  createMappings(data) {
+    const mappings = [];
+
+    if (data['radius']) {
+      const rm = new RadiusMapping();
+      rm.field = data['radius'].name || data['radius'];
+      rm.setUnits(data['radiusUnits']);
+      mappings.push(rm);
+    }
+    if (data['semiMajor']) {
+      var smaj = new SemiMajorMapping();
+      smaj.field = data['semiMajor'].name || data['semiMajor'];
+      smaj.setUnits(data['semiMajorUnits']);
+      mappings.push(smaj);
+    }
+    if (data['semiMinor']) {
+      var smin = new SemiMinorMapping();
+      smin.field = data['semiMinor'].name || data['semiMinor'];
+      smin.setUnits(data['semiMinorUnits']);
+      mappings.push(smin);
+    }
+    if (data['orientation']) {
+      var om = new OrientationMapping();
+      om.field = data['orientation'].name || data['orientation'];
+      mappings.push(om);
+    }
+
+
+    if (this.source_) {
+      if (!this.sourceListener_) this.listenerSetup_();
+      this.setMappings(mappings);
+    }
+
+    return mappings;
+  }
+
+
+  /**
+   * Returns the mappings applied to the layer
+   * @return {Array<RenameMapping>}
+   */
+  getMappings() {
+    return this.source_.get(EllipseMappingKey);
+  }
+
+
+  /**
+   * Returns the mappings applied to the layer
+   * @param {Array<RenameMapping>} value
+   */
+  setMappings(value) {
+    this.source_.set(EllipseMappingKey, value, true);
+  }
+
+
+  /**
+   * Gets the mapping options for later use
+   * @return {MappingOptions|undefined}
+   */
+  getMappingOptions() {
+    return this.mappingOptions_;
+  }
+
+
+  /**
+   * Sets the mapping options for later use
+   * @param {MappingOptions|undefined} value
+   */
+  setMappingOptions(value) {
+    this.mappingOptions_ = value;
+  }
+
+
+
+  /**
+   * Execute the Mappings for the source
+   */
+  executeMappings() {
+    const sourceId = this.source_.getId();
+    const layer = os.map.mapContainer.getLayer(sourceId);
```

Comment:
goog.require() and do the getInstance()

Reply:
This one is going to have to stay like that for now. Using `os.MapContainer` results in a circular dependency. I went ahead and added a comment on it though.